Pytest deprecation breaking tests

<!--
**Reminders**
1. This is not a place to debug your MCNP models
1. MCNP is export controlled and only its 6.2 and 6.3 user manuals are public. Don't include any information about MCNP which is not in those manuals.
1. Your model may be export controlled, or proprietary. Please change specific numbers (e.g., dimensions, material compositions, etc.) in your minimum working examples. -->

## Bug Description

A recent release of Pytest, 9.1.0, deprecated class scope fixtures: https://github.com/pytest-dev/pytest/releases/tag/9.1.0, which I think broke our tests: https://github.com/idaholab/MontePy/actions/runs/27505921402/job/81296994783?pr=945.  

## To Reproduce

<!-- A short code snippet of what you have ran. Please change or remove any specific values or anything that can't be public. For example: --> 

``` shell
pip install --upgrade pytest
pytest
```

## Error Message 

<!-- If an error message was printed please include the entire stacktrace. If it includes any specific values please change or remove them. For example: -->

<details open> 

``` python
self = <FixtureDef argname='m0_prob' scope='class' baseid='tests/test_numbered_collection.py::TestMaterials'>
request = <SubRequest 'm0_prob' for <Function test_get_containing_any[nuclides0-1.0-1]>>

    def execute(self, request: SubRequest) -> FixtureValue:
        """Return the value of this fixture, executing it if not cached."""
        # Ensure that the dependent fixtures requested by this fixture are loaded.
        # This needs to be done before checking if we have a cached value, since
        # if a dependent fixture has their cache invalidated, e.g. due to
        # parametrization, they finalize themselves and fixtures depending on it
        # (which will likely include this fixture) setting `self.cached_result = None`.
        # See #4871
        requested_fixtures_that_should_finalize_us = []
        for argname in self.argnames:
            fixturedef = request._get_active_fixturedef(argname)
            # Saves requested fixtures in a list so we later can add our finalizer
            # to them, ensuring that if a requested fixture gets torn down we get torn
            # down first. This is generally handled by SetupState, but still currently
            # needed when this fixture is not parametrized but depends on a parametrized
            # fixture.
            requested_fixtures_that_should_finalize_us.append(fixturedef)

        # Check for (and return) cached value/exception.
        if self.cached_result is not None:
            request_cache_key = self.cache_key(request)
            cache_key = self.cached_result[1]
            try:
                # Attempt to make a normal == check: this might fail for objects
                # which do not implement the standard comparison (like numpy arrays -- #6497).
                cache_hit = bool(request_cache_key == cache_key)
            except (ValueError, RuntimeError):
                # If the comparison raises, use 'is' as fallback.
                cache_hit = request_cache_key is cache_key

            if cache_hit:
                if self.cached_result[2] is not None:
                    exc, exc_tb = self.cached_result[2]
                    raise exc.with_traceback(exc_tb)
                else:
                    return self.cached_result[0]
            # We have a previous but differently parametrized fixture instance
            # so we need to tear it down before creating a new one.
            self.finish(request)
            assert self.cached_result is None

        # Add finalizer to requested fixtures we saved previously.
        # We make sure to do this after checking for cached value to avoid
        # adding our finalizer multiple times. (#12135)
        finalizer = functools.partial(self.finish, request=request)
        for parent_fixture in requested_fixtures_that_should_finalize_us:
            parent_fixture.addfinalizer(finalizer)

        # Register the pytest_fixture_post_finalizer as the first finalizer,
        # which is executed last.
>       assert not self._finalizers
               ^^^^^^^^^^^^^^^^^^^^
E       AssertionError

../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:1220: AssertionError
_______________________________________________ ERROR at setup of TestMaterials.test_get_containing_any[nuclides1-0.0-1] _______________________________________________

cls = <class '_pytest.runner.CallInfo'>, func = <function call_and_report.<locals>.<lambda> at 0x7845ce208b40>, when = 'setup'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: Callable[[], TResult],
        when: Literal["collect", "setup", "call", "teardown"],
        reraise: type[BaseException] | tuple[type[BaseException], ...] | None = None,
    ) -> CallInfo[TResult]:
        """Call func, wrapping the result in a CallInfo.

        :param func:
            The function to call. Called without arguments.
        :type func: Callable[[], _pytest.runner.TResult]
        :param when:
            The phase in which the function is called.
        :param reraise:
            Exception or exceptions that shall propagate if raised by the
            function, instead of being wrapped in the CallInfo.
        """
        excinfo = None
        instant = timing.Instant()
        try:
>           result: TResult | None = func()
                                     ^^^^^^

../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:361:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:250: in <lambda>
    lambda: runtest_hook(item=item, **kwds),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../.local/lib/python3.14/site-packages/pluggy/_hooks.py:512: in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../.local/lib/python3.14/site-packages/pluggy/_manager.py:120: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/logging.py:858: in pytest_runtest_setup
    yield
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/capture.py:895: in pytest_runtest_setup
    return (yield)
            ^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:170: in pytest_runtest_setup
    item.session._setupstate.setup(item)
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:536: in setup
    col.setup()
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/python.py:1710: in setup
    self._request._fillfixtures()
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:805: in _fillfixtures
    item.funcargs[argname] = self.getfixturevalue(argname)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:629: in getfixturevalue
    fixturedef = self._get_active_fixturedef(argname)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:725: in _get_active_fixturedef
    fixturedef.execute(request=subrequest)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <FixtureDef argname='m0_prob' scope='class' baseid='tests/test_numbered_collection.py::TestMaterials'>
request = <SubRequest 'm0_prob' for <Function test_get_containing_any[nuclides1-0.0-1]>>

    def execute(self, request: SubRequest) -> FixtureValue:
        """Return the value of this fixture, executing it if not cached."""
        # Ensure that the dependent fixtures requested by this fixture are loaded.
        # This needs to be done before checking if we have a cached value, since
        # if a dependent fixture has their cache invalidated, e.g. due to
        # parametrization, they finalize themselves and fixtures depending on it
        # (which will likely include this fixture) setting `self.cached_result = None`.
        # See #4871
        requested_fixtures_that_should_finalize_us = []
        for argname in self.argnames:
            fixturedef = request._get_active_fixturedef(argname)
            # Saves requested fixtures in a list so we later can add our finalizer
            # to them, ensuring that if a requested fixture gets torn down we get torn
            # down first. This is generally handled by SetupState, but still currently
            # needed when this fixture is not parametrized but depends on a parametrized
            # fixture.
            requested_fixtures_that_should_finalize_us.append(fixturedef)

        # Check for (and return) cached value/exception.
        if self.cached_result is not None:
            request_cache_key = self.cache_key(request)
            cache_key = self.cached_result[1]
            try:
                # Attempt to make a normal == check: this might fail for objects
                # which do not implement the standard comparison (like numpy arrays -- #6497).
                cache_hit = bool(request_cache_key == cache_key)
            except (ValueError, RuntimeError):
                # If the comparison raises, use 'is' as fallback.
                cache_hit = request_cache_key is cache_key

            if cache_hit:
                if self.cached_result[2] is not None:
                    exc, exc_tb = self.cached_result[2]
                    raise exc.with_traceback(exc_tb)
                else:
                    return self.cached_result[0]
            # We have a previous but differently parametrized fixture instance
            # so we need to tear it down before creating a new one.
            self.finish(request)
            assert self.cached_result is None

        # Add finalizer to requested fixtures we saved previously.
        # We make sure to do this after checking for cached value to avoid
        # adding our finalizer multiple times. (#12135)
        finalizer = functools.partial(self.finish, request=request)
        for parent_fixture in requested_fixtures_that_should_finalize_us:
            parent_fixture.addfinalizer(finalizer)

        # Register the pytest_fixture_post_finalizer as the first finalizer,
        # which is executed last.
>       assert not self._finalizers
               ^^^^^^^^^^^^^^^^^^^^
E       AssertionError

../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:1220: AssertionError
_______________________________________________ ERROR at setup of TestMaterials.test_get_containing_any[nuclides2-1.0-0] _______________________________________________

cls = <class '_pytest.runner.CallInfo'>, func = <function call_and_report.<locals>.<lambda> at 0x7845ce208040>, when = 'setup'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: Callable[[], TResult],
        when: Literal["collect", "setup", "call", "teardown"],
        reraise: type[BaseException] | tuple[type[BaseException], ...] | None = None,
    ) -> CallInfo[TResult]:
        """Call func, wrapping the result in a CallInfo.

        :param func:
            The function to call. Called without arguments.
        :type func: Callable[[], _pytest.runner.TResult]
        :param when:
            The phase in which the function is called.
        :param reraise:
            Exception or exceptions that shall propagate if raised by the
            function, instead of being wrapped in the CallInfo.
        """
        excinfo = None
        instant = timing.Instant()
        try:
>           result: TResult | None = func()
                                     ^^^^^^

../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:361:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:250: in <lambda>
    lambda: runtest_hook(item=item, **kwds),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../.local/lib/python3.14/site-packages/pluggy/_hooks.py:512: in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../.local/lib/python3.14/site-packages/pluggy/_manager.py:120: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/logging.py:858: in pytest_runtest_setup
    yield
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/capture.py:895: in pytest_runtest_setup
    return (yield)
            ^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:170: in pytest_runtest_setup
    item.session._setupstate.setup(item)
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:536: in setup
    col.setup()
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/python.py:1710: in setup
    self._request._fillfixtures()
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:805: in _fillfixtures
    item.funcargs[argname] = self.getfixturevalue(argname)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:629: in getfixturevalue
    fixturedef = self._get_active_fixturedef(argname)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:725: in _get_active_fixturedef
    fixturedef.execute(request=subrequest)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <FixtureDef argname='m0_prob' scope='class' baseid='tests/test_numbered_collection.py::TestMaterials'>
request = <SubRequest 'm0_prob' for <Function test_get_containing_any[nuclides2-1.0-0]>>

    def execute(self, request: SubRequest) -> FixtureValue:
        """Return the value of this fixture, executing it if not cached."""
        # Ensure that the dependent fixtures requested by this fixture are loaded.
        # This needs to be done before checking if we have a cached value, since
        # if a dependent fixture has their cache invalidated, e.g. due to
        # parametrization, they finalize themselves and fixtures depending on it
        # (which will likely include this fixture) setting `self.cached_result = None`.
        # See #4871
        requested_fixtures_that_should_finalize_us = []
        for argname in self.argnames:
            fixturedef = request._get_active_fixturedef(argname)
            # Saves requested fixtures in a list so we later can add our finalizer
            # to them, ensuring that if a requested fixture gets torn down we get torn
            # down first. This is generally handled by SetupState, but still currently
            # needed when this fixture is not parametrized but depends on a parametrized
            # fixture.
            requested_fixtures_that_should_finalize_us.append(fixturedef)

        # Check for (and return) cached value/exception.
        if self.cached_result is not None:
            request_cache_key = self.cache_key(request)
            cache_key = self.cached_result[1]
            try:
                # Attempt to make a normal == check: this might fail for objects
                # which do not implement the standard comparison (like numpy arrays -- #6497).
                cache_hit = bool(request_cache_key == cache_key)
            except (ValueError, RuntimeError):
                # If the comparison raises, use 'is' as fallback.
                cache_hit = request_cache_key is cache_key

            if cache_hit:
                if self.cached_result[2] is not None:
                    exc, exc_tb = self.cached_result[2]
                    raise exc.with_traceback(exc_tb)
                else:
                    return self.cached_result[0]
            # We have a previous but differently parametrized fixture instance
            # so we need to tear it down before creating a new one.
            self.finish(request)
            assert self.cached_result is None

        # Add finalizer to requested fixtures we saved previously.
        # We make sure to do this after checking for cached value to avoid
        # adding our finalizer multiple times. (#12135)
        finalizer = functools.partial(self.finish, request=request)
        for parent_fixture in requested_fixtures_that_should_finalize_us:
            parent_fixture.addfinalizer(finalizer)

        # Register the pytest_fixture_post_finalizer as the first finalizer,
        # which is executed last.
>       assert not self._finalizers
               ^^^^^^^^^^^^^^^^^^^^
E       AssertionError

../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:1220: AssertionError
_______________________________________________ ERROR at setup of TestMaterials.test_get_containing_any[nuclides3-0.0-2] _______________________________________________

cls = <class '_pytest.runner.CallInfo'>, func = <function call_and_report.<locals>.<lambda> at 0x7845ce208510>, when = 'setup'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: Callable[[], TResult],
        when: Literal["collect", "setup", "call", "teardown"],
        reraise: type[BaseException] | tuple[type[BaseException], ...] | None = None,
    ) -> CallInfo[TResult]:
        """Call func, wrapping the result in a CallInfo.

        :param func:
            The function to call. Called without arguments.
        :type func: Callable[[], _pytest.runner.TResult]
        :param when:
            The phase in which the function is called.
        :param reraise:
            Exception or exceptions that shall propagate if raised by the
            function, instead of being wrapped in the CallInfo.
        """
        excinfo = None
        instant = timing.Instant()
        try:
>           result: TResult | None = func()
                                     ^^^^^^

../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:361:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:250: in <lambda>
    lambda: runtest_hook(item=item, **kwds),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../.local/lib/python3.14/site-packages/pluggy/_hooks.py:512: in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../.local/lib/python3.14/site-packages/pluggy/_manager.py:120: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/logging.py:858: in pytest_runtest_setup
    yield
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/capture.py:895: in pytest_runtest_setup
    return (yield)
            ^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:170: in pytest_runtest_setup
    item.session._setupstate.setup(item)
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/runner.py:536: in setup
    col.setup()
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/python.py:1710: in setup
    self._request._fillfixtures()
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:805: in _fillfixtures
    item.funcargs[argname] = self.getfixturevalue(argname)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:629: in getfixturevalue
    fixturedef = self._get_active_fixturedef(argname)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:725: in _get_active_fixturedef
    fixturedef.execute(request=subrequest)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <FixtureDef argname='m0_prob' scope='class' baseid='tests/test_numbered_collection.py::TestMaterials'>
request = <SubRequest 'm0_prob' for <Function test_get_containing_any[nuclides3-0.0-2]>>

    def execute(self, request: SubRequest) -> FixtureValue:
        """Return the value of this fixture, executing it if not cached."""
        # Ensure that the dependent fixtures requested by this fixture are loaded.
        # This needs to be done before checking if we have a cached value, since
        # if a dependent fixture has their cache invalidated, e.g. due to
        # parametrization, they finalize themselves and fixtures depending on it
        # (which will likely include this fixture) setting `self.cached_result = None`.
        # See #4871
        requested_fixtures_that_should_finalize_us = []
        for argname in self.argnames:
            fixturedef = request._get_active_fixturedef(argname)
            # Saves requested fixtures in a list so we later can add our finalizer
            # to them, ensuring that if a requested fixture gets torn down we get torn
            # down first. This is generally handled by SetupState, but still currently
            # needed when this fixture is not parametrized but depends on a parametrized
            # fixture.
            requested_fixtures_that_should_finalize_us.append(fixturedef)

        # Check for (and return) cached value/exception.
        if self.cached_result is not None:
            request_cache_key = self.cache_key(request)
            cache_key = self.cached_result[1]
            try:
                # Attempt to make a normal == check: this might fail for objects
                # which do not implement the standard comparison (like numpy arrays -- #6497).
                cache_hit = bool(request_cache_key == cache_key)
            except (ValueError, RuntimeError):
                # If the comparison raises, use 'is' as fallback.
                cache_hit = request_cache_key is cache_key

            if cache_hit:
                if self.cached_result[2] is not None:
                    exc, exc_tb = self.cached_result[2]
                    raise exc.with_traceback(exc_tb)
                else:
                    return self.cached_result[0]
            # We have a previous but differently parametrized fixture instance
            # so we need to tear it down before creating a new one.
            self.finish(request)
            assert self.cached_result is None

        # Add finalizer to requested fixtures we saved previously.
        # We make sure to do this after checking for cached value to avoid
        # adding our finalizer multiple times. (#12135)
        finalizer = functools.partial(self.finish, request=request)
        for parent_fixture in requested_fixtures_that_should_finalize_us:
            parent_fixture.addfinalizer(finalizer)

        # Register the pytest_fixture_post_finalizer as the first finalizer,
        # which is executed last.
>       assert not self._finalizers
               ^^^^^^^^^^^^^^^^^^^^
E       AssertionError

../../miniforge3/envs/montepy/lib/python3.14/site-packages/_pytest/fixtures.py:1220: AssertionError
=============================================================================== FAILURES ===============================================================================
___________________________________________________________________ TestFill.test_fill_index_setter ____________________________________________________________________

self = <tests.test_universe.TestFill object at 0x7845cf6f84b0>

    @given(
>       indices=st.lists(st.integers(), min_size=3, max_size=3),
               ^^^^^^^
        width=st.lists(st.integers(1), min_size=3, max_size=3),
    )

tests/test_universe.py:492:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_universe.py:500: in test_fill_index_setter
    fill.max_index = end
    ^^^^^^^^^^^^^^
montepy/mcnp_object.py:49: in wrapped
    add_line_number_to_exception(e, self)
montepy/exceptions.py:235: in add_line_number_to_exception
    raise error
montepy/mcnp_object.py:41: in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
montepy/mcnp_object.py:150: in __setattr__
    descriptor.__set__(self, value)
montepy/mcnp_object.py:49: in wrapped
    add_line_number_to_exception(e, self)
montepy/exceptions.py:259: in add_line_number_to_exception
    raise error.with_traceback(trace)
montepy/mcnp_object.py:41: in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
montepy/utilities.py:181: in setter
    validator(self, value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Fill: set_in_cell: True, in_cell: True old_number: 5, old_transform: None old_numbers: None Universe: None, universes: None, transform: None Multi_universe: True hidden_transform: None Min/Max: [0, 0, 9223372036854775808] None
indices = array([1.00000000e+00, 1.00000000e+00, 9.22337204e+18])

    def _verify_3d_index(self, indices):
        for index in indices:
            if not isinstance(index, Integral):
>               raise TypeError(f"Index values for fill must be an int. {index} given.")
E               TypeError: Index values for fill must be an int. 1.0 given.
E
E               Error came from Fill: Universe: None, transform: None from an unknown file.
E               Falsifying example: test_fill_index_setter(
E                   self=<tests.test_universe.TestFill object at 0x7845cf6f84b0>,
E                   indices=[0, 0, 9_223_372_036_854_775_808],
E                   width=[1, 1, 1],
E               )

montepy/data_inputs/fill.py:20: TypeError
======================================================================= short test summary info ========================================================================
FAILED tests/test_universe.py::TestFill::test_fill_index_setter - TypeError: Index values for fill must be an int. 1.0 given.
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_numbers - pytest.PytestRemovedIn10Warning: Class-scoped fixture defined as instance method is deprecated.
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_number_conflict_init - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_check_number - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_objects - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_pop - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_extend - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_iter - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_append - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_append_renumber - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_append_renumber_none_number - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_append_renumber_problems - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_extend_renumber - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_request_number - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_next_number - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_getitem - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_delete - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_setitem - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_iadd - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_slice - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_get - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_keys - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_values - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_items - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_eq - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_surface_generators - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_number_adding_concurancy - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_str - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_data_init - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_data_append - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_data_append_renumber - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_data_remove - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_numbered_discard - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_numbered_contains - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_data_delete - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_data_clear - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_data_pop - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_num_collect_clone - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_cells_clone - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_num_collect_clone_default - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_num_collect_link_problem - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_num_collect_clone_bad[args0-TypeError] - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_num_collect_clone_bad[args1-TypeError] - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_num_collect_clone_bad[args2-ValueError] - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_num_collect_clone_bad[args3-ValueError] - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_num_collect_clone_bad[args4-ValueError] - AssertionError
ERROR tests/test_numbered_collection.py::TestNumberedObjectCollection::test_num_collect_clone_bad[args5-ValueError] - AssertionError
ERROR tests/test_numbered_collection.py::TestMaterials::test_m0_defaults - pytest.PytestRemovedIn10Warning: Class-scoped fixture defined as instance method is deprecated.
ERROR tests/test_numbered_collection.py::TestMaterials::test_get_containing_all[nuclides0-1.0-1] - AssertionError
ERROR tests/test_numbered_collection.py::TestMaterials::test_get_containing_all[nuclides1-0.0-1] - AssertionError
ERROR tests/test_numbered_collection.py::TestMaterials::test_get_containing_all[nuclides2-1.0-0] - AssertionError
ERROR tests/test_numbered_collection.py::TestMaterials::test_get_containing_any[nuclides0-1.0-1] - AssertionError
ERROR tests/test_numbered_collection.py::TestMaterials::test_get_containing_any[nuclides1-0.0-1] - AssertionError
ERROR tests/test_numbered_collection.py::TestMaterials::test_get_containing_any[nuclides2-1.0-0] - AssertionError
ERROR tests/test_numbered_collection.py::TestMaterials::test_get_containing_any[nuclides3-0.0-2] - AssertionError
============================================================== 1 failed, 1160 passed, 54 errors in 11.69s ==============================================================
```

</details>


## Version

 - Version 1.4.1dev...

## Additional context

Add any other context about the problem here.

--- a/tests/test_numbered_collection.py
+++ b/tests/test_numbered_collection.py
@@ -13,7 +13,8 @@ import os
 
 class TestNumberedObjectCollection:
 
-    @pytest.fixture(scope="class")
+    @pytest.fixture
+    @classmethod
     def read_simple_problem(_):
         return montepy.read_input(os.path.join("tests", "inputs", "test.imcnp"))
 
@@ -825,7 +826,8 @@ class TestNumberedObjectCollection:
 
 class TestMaterials:
 
-    @pytest.fixture(scope="class")
+    @pytest.fixture
+    @classmethod
     def m0_prob(_):
         return montepy.read_input(
             os.path.join("tests", "inputs", "test_importance.imcnp")

--- a/tests/test_universe.py
+++ b/tests/test_universe.py
@@ -489,8 +489,8 @@ class TestFill:
             fill1.merge(fill2)
 
     @given(
-        indices=st.lists(st.integers(), min_size=3, max_size=3),
-        width=st.lists(st.integers(1), min_size=3, max_size=3),
+        indices=st.lists(st.integers(-1_000_000, 1_000_000), min_size=3, max_size=3),
+        width=st.lists(st.integers(1, 1_000_000), min_size=3, max_size=3),
     )
     def test_fill_index_setter(self, indices, width):
         fill = self.simple_fill.clone()

